src/System.Runtime.Extensions/tests/System/AppDomainTests.netcoreapp.cs

```diff
@@ -16,6 +21,168 @@ public void GetSetupInformation()
+
+                obj = AppDomain.CurrentDomain.CreateInstanceFromAndUnwrap(assemblyFile: assemblyFile, typeName: type, null);
+                Assert.NotNull(obj);
+                Assert.Equal(returnedFullNameType, obj.GetType().FullName);
+            }
+            Assert.True(File.Exists(physicalFileName));
+        }
+
+        public static IEnumerable<object[]> TestingCreateInstanceFromObjectHandle()
+        {
+            //string physicalFileName, string assemblyFile, string typeName, returnedFullNameType, expectedException
+            yield return new object[] { "AssemblyResolveTests.dll", "AssemblyResolveTests.dll", "AssemblyResolveTests.PublicClassSample", "AssemblyResolveTests.PublicClassSample", null };
+            yield return new object[] { "AssemblyResolveTests.dll", "assemblyresolvetests.dll", "assemblyresolvetests.publicclasssample", "AssemblyResolveTests.PublicClassSample", typeof(TypeLoadException) };
+
+            yield return new object[] { "AssemblyResolveTests.dll", "AssemblyResolveTests.dll", "AssemblyResolveTests.PrivateClassSample", "AssemblyResolveTests.PrivateClassSample", null };
+            yield return new object[] { "AssemblyResolveTests.dll", "assemblyresolvetests.dll", "assemblyresolvetests.privateclasssample", "AssemblyResolveTests.PrivateClassSample", typeof(TypeLoadException) };
+
+            yield return new object[] { "AssemblyResolveTests.dll", "AssemblyResolveTests.dll", "AssemblyResolveTests.PublicClassNoDefaultConstructorSample", "AssemblyResolveTests.PublicClassNoDefaultConstructorSample", typeof(MissingMethodException) };
+            yield return new object[] { "AssemblyResolveTests.dll", "assemblyresolvetests.dll", "assemblyresolvetests.publicclassnodefaultconstructorsample", "AssemblyResolveTests.PublicClassNoDefaultConstructorSample", typeof(TypeLoadException) };
+        }
+
+        [Theory]
+        [MemberData(nameof(TestingCreateInstanceObjectHandle))]
+        static void TestingCreateInstanceObjectHandle(string assemblyName, string type, string returnedFullNameType, Type exceptionType)
+        {
+            ObjectHandle oh = null;
+            object obj = null;
+
+            if (exceptionType != null)
+            {
+                Assert.Throws(exceptionType, () => AppDomain.CurrentDomain.CreateInstance(assemblyName: assemblyName, typeName: type));
+                Assert.Throws(exceptionType, () => AppDomain.CurrentDomain.CreateInstanceAndUnwrap(assemblyName: assemblyName, typeName: type));
+            }
+            else
+            {
+                oh = AppDomain.CurrentDomain.CreateInstance(assemblyName: assemblyName, typeName: type);
+                Assert.NotNull(oh);
+                Assert.Equal(returnedFullNameType, oh.Unwrap().GetType().FullName);
+
+                obj = AppDomain.CurrentDomain.CreateInstanceAndUnwrap(assemblyName: assemblyName, typeName: type);
+                Assert.NotNull(obj);
```

Comment:
can factor out these two lines into private function something like ```CheckValidity```. will make the repetition less

Reply:
done